app/scripts/modules/openstack/securityGroup/configure/wizard/rules.html

```diff
@@ -0,0 +1,42 @@
+<form ng-controller="openstackSecurityGroupRulesController as rulesController" class="container-fluid form-horizontal" name="rules">
+  <div class="form-group">
+    <div class="col-md-12">
+      <table class="table table-condensed packed">
+        <thead>
+        <tr>
+          <th style="width: 15%">Protocol</th>
+          <th style="width: 15%">From Port</th>
+          <th style="width: 15%">To Port</th>
+          <th style="width: 15%">CIDR</th>
+          <th></th>
+        </tr>
+        </thead>
+        <div>
+          <tbody>
+            <tr ng-repeat="rule in securityGroup.rules">
+              <td><select class="form-control input-sm" ng-model="rule.type" ng-init="rule.type='tcp'"
+                          ng-options="protocol as protocol.toUpperCase() for protocol in ['tcp']" required></select></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.fromPort=80" min="0" ng-model="rule.fromPort" required/></td>
+              <td><input class="form-control input-sm" type="number" ng-init="rule.toPort=80" min="0" ng-model="rule.toPort" required/></td>
```

Comment:
I did the same thing first, but I was not able to display validation message, so though to take same different approach.
Hence added TODO.

I will again do the change to replace 0 with rule.fromPort and try to display validation message. If validation message is not possible can we omit that message.